x-pack/plugins/security_solution/public/detections/components/alerts_table/actions.tsx

```diff
@@ -146,83 +150,98 @@ const getFiltersFromRule = (filters: string[]): Filter[] =>
+    (outerAcc, thresholdData) => {
+      const threshold = thresholdData.signal?.rule?.threshold as string[];
+
+      let aggField: string[] = [];
+      let thresholdResult: {
+        terms?: Array<{
+          field?: string;
+          value: string;
+        }>;
+        count: number;
+        from: string;
       };
-    }
-
-    const aggregationFields = Array.isArray(aggField) ? aggField : [aggField];
-
-    return [
-      ...outerAcc,
-      ...aggregationFields.reduce<DataProvider[]>((acc, aggregationField, i) => {
-        const aggregationValue = (thresholdResult.terms ?? []).filter(
-          (term: { field?: string | undefined; value: string }) => term.field === aggregationField
-        )[0].value;
-        const dataProviderValue = Array.isArray(aggregationValue)
-          ? aggregationValue[0]
-          : aggregationValue;
-
-        if (!dataProviderValue) {
-          return acc;
-        }
-
-        const aggregationFieldId = aggregationField.replace('.', '-');
-        const dataProviderPartial = {
-          id: `send-alert-to-timeline-action-default-draggable-event-details-value-formatted-field-value-${TimelineId.active}-${aggregationFieldId}-${dataProviderValue}`,
-          name: aggregationField,
-          enabled: true,
-          excluded: false,
-          kqlQuery: '',
-          queryMatch: {
-            field: aggregationField,
-            value: dataProviderValue,
-            operator: ':' as QueryOperator,
-          },
-        };
 
-        if (i === 0) {
-          return [
-            ...acc,
+      try {
+        thresholdResult = JSON.parse((thresholdData.signal?.threshold_result as string[])[0]);
+        aggField = JSON.parse(threshold[0]).field;
+      } catch (err) {
+        thresholdResult = {
+          terms: [
             {
-              ...dataProviderPartial,
-              and: [],
+              field: (thresholdData.rule?.threshold as { field: string }).field,
+              value: (thresholdData.signal?.threshold_result as { value: string }).value,
             },
-          ];
-        } else {
-          acc[0].and.push(dataProviderPartial);
-          return acc;
-        }
-      }, []),
-    ];
-  }, []);
+          ],
+          count: (thresholdData.signal?.threshold_result as { count: number }).count,
+          from: (thresholdData.signal?.threshold_result as { from: string }).from,
+        };
+      }
+
+      const originalTime = moment(thresholdData.signal?.original_time![0]);
+      const now = moment();
+      const ruleFrom = dateMath.parse(thresholdData.signal?.rule?.from![0]!);
+      const ruleInterval = moment.duration(now.diff(ruleFrom));
+      const fromOriginalTime = originalTime.clone().subtract(ruleInterval); // This is the default... can overshoot
+      const aggregationFields = Array.isArray(aggField) ? aggField : [aggField];
+
+      return {
+        thresholdFrom: thresholdResult.from ?? fromOriginalTime.toISOString(),
```

Comment:
Use `threshold_result.from` if available (it will always be available for new signals). Otherwise, use a calculated lower bound, which could result in the timeline showing a superset of the events that made up the threshold set.